builders/build/canonical-contracts/precompiles/batch.md

```diff
@@ -0,0 +1,237 @@
+
+```
+["0x6Be02d1d3665660d22FF9624b7BE0551ee1Ac91b", "0xd14b70a55F6cBAc06d4FA49b99be0370D0e1BD39", "0xd14b70a55F6cBAc06d4FA49b99be0370D0e1BD39"]
+``` 
+
+There will also be 3 values for the *value* array. The first address in the *to* input array has to do with sending 1 DEV, so 1 DEV in Wei is within the array. The following two values are 0 because the function that their subtransactions are interacting with do not accept or require native currency.  
+
+```
+["1000000000000000000", "0", "0"]
+```
+
+We will need 3 values for the *call_data* array. Since transferring native currency does not require call data, the string is simply blank. The second and third values in the array correspond to invokations of **setMessage** that set messages to ids 5 and 6.
+
+```
+["", "0x648345c8000000000000000000000000000000000000000000000000000000000000000500000000000000000000000000000000000000000000000000000000000000400000000000000000000000000000000000000000000000000000000000000009796f752061726520610000000000000000000000000000000000000000000000", "0x648345c800000000000000000000000000000000000000000000000000000000000000060000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000e61206d6f6f6e6265616d2070726f000000000000000000000000000000000000"]
+```
+
+The final input is for *gas_input*. This array will be left empty to forward all remaining gas to each subtransaction.
+
+```
+[]
+```
+
+Try sending a batched transaction with these inputs in Remix the same way [you did before](#function-interaction-via-precompile).
+
+And that's it! You've successfully interacted with the ERC-20 precompile using MetaMask and Remix!
+
+## Using Ethereum Developer Libraries {: #using-ethereum-developer-libraries }
+
+If you have only followed the [Ethers.js tutorial](/builders/build/eth-api/libraries/ethersjs/) on Moonbeam, you may find it difficult to find the call data for a function. The answer is hidden within Ether's `utils.Interface` object, where the [encodeFunctionData](https://docs.ethers.io/v5/api/utils/abi/interface/#Interface--encoding){target=_blank} function allows you to input your function name and inputs to receive the resultant call data. Web3js has a similar function, [encodeFunctionCall](https://web3js.readthedocs.io/en/v1.2.11/web3-eth-abi.html#encodefunctioncall){target=_blank}.
+
+!!! note
+    The code snippets presented in the following sections are not meant for production environments. Please make sure you adapt it for each use-case.
+
+=== "web3.js"
+     --8<-- 'code/batch/web3js-batch.md'
+
+=== "ethers.js"
+     --8<-- 'code/batch/ethers-batch.md'
+
+=== "web3.py"
+     --8<-- 'code/batch/web3py-batch.md'
```

Comment:
these tab titles can all be capitalized (created a ticket for standardizing this throughout docs cause I know it's different elsewhere)